samcli/local/docker/container.py

```diff
@@ -157,6 +168,15 @@ def create(self):
             "use_config_proxy": True,
         }
 
+        # Get effective user when building lambda and mounting with write permissions
+        # Pass effective user to docker run CLI as "--user" option in the format of uid[:gid]
+        # to run docker as current user instead of root
+        # Skip if current user is root on posix systems or non-posix systems
+        _root_user_id = "0"
+        effective_user = EffectiveUser.get_current_effective_user().to_effective_user_str()
+        if self._mount_with_write and effective_user and effective_user != _root_user_id:
```

Comment:
Can we add some debug logs here to investigate any possible issue in the future?